enum.rb

```diff
@@ -0,0 +1,186 @@
+module Enumerable
+  def my_each
+    return Enumerator unless block_given?
+
+    ar = self
+    ar.each do |el|
+      yield el
+    end
+  end
+
+  def my_each_with_index
+    return Enumerator unless block_given?
+
+    i = 0
+    ar = self
+    ar.my_each do |element|
+      yield i, element
+      i += 1
+    end
+  end
```

Comment:
* [ ] should have working `my_each_with_index` identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
my_each_output = 0
my_each_output2 = 0
[1,2,3,4,5].my_each_with_index{ |num, idx| my_each_output += "Num: #{num}, idx: #{idx}\n" }
[1,2,3,4,5].each_with_index{ |num, idx| my_each_output2 += "Num: #{num}, idx: #{idx}\n" }
my_each_output != my_each_output2

[1,2,3,4,5].my_each_with_index != [1,2,3,4,5].each_with_index
```